src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -91,6 +91,74 @@ public class TestExtensions
     /// </summary>
     internal bool AreDataCollectorsCached { get; set; }
 
+    /// <summary>
+    /// Merge two extension maps.
+    /// </summary>
+    /// 
+    /// <param name="extensionMap1">First extension map.</param>
+    /// <param name="extensionMap2">Second extension map.</param>
+    /// 
+    /// <returns>
+    /// A map representing the merger between the two input maps.
+    /// </returns>
+    internal static IDictionary<string, ISet<string>> MergeExtensionMaps(
```

Comment:
[minor] "Map" feels javaish to me, I would use dictionary.

Reply:
Agree. Reading that description comment and the name I have no idea what this is.

Reply:
Taking all input into consideration, I renamed the method `MergeDictionaries`.

`MergeByKey` gives away implementation details;
`MergeMaps`, both Jakub and Amaury think the term `map` should be replaced by `dictionary` to make the operation more self descriptive;
`Merge`/`Combine` is too generic and implies it merges two `TestExtension` objects together which is not really the case, we merge two dictionaries that happen to contain extensions.